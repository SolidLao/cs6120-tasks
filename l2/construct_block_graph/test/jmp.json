{
  "functions": [
    {
      "name": "main",
      "instrs": [
        {
          "dest": "v",
          "op": "const",
          "type": "int",
          "value": 4
        },
        {
          "op": "jmp",
          "labels": ["somewhere"]
        },
        {
          "dest": "v",
          "op": "const",
          "type": "int",
          "value": 2
        },
        {
          "label": "somewhere"
        },
        {
          "args": ["v"],
          "op": "print"
        }
      ]
    }
  ]
}