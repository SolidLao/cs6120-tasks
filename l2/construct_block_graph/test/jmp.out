Basic Blocks:
========================================

Block 'b1':
  v = const 4
  jmp somewhere

Block 'b2':
  v = const 2
  jmp somewhere

Block 'somewhere':
  print v
  ret

Control Flow Graph (main):
========================================
b1 -> somewhere
b2 -> somewhere
somewhere -> exit
